addons/website_sale/static/src/js/checkout.js

```diff
@@ -1,70 +1,459 @@
 import { rpc } from "@web/core/network/rpc";
 
 publicWidget.registry.websiteSaleCheckout = publicWidget.Widget.extend({
     // /shop/checkout
     selector: '#shop_checkout',
     events: {
-        'click .js_change_billing': '_onClickChangeBilling',
-        'click .js_change_shipping': '_onClickChangeShipping',
-        'click .js_edit_address': '_onClickEditAddress',
+        // addresses
+        'click .js_change_billing': '_changeBillingAddress',
+        'click .js_change_shipping': '_changeShippingAddress',
+        'click .js_edit_address': '_editAddress',
+        // delivery
+        'click [name="o_delivery_radio"]': '_selectDeliveryCarrier',
+        'click .o_address_select': '_selectLocation',
+        'click .o_remove_order_location': '_removeLocation',
+    },
+
+    async start(){
+        this.mainButton = document.querySelector('a[name="website_sale_main_button"]');
+        await this._prepareDeliveryCarriers();
     },
 
     //--------------------------------------------------------------------------
-    // Handlers
+    // Event handlers
     //--------------------------------------------------------------------------
 
     /**
      * @private
      * @param {Event} ev
      */
-    _onClickChangeBilling: function (ev) {
-        this._onClickChangeAddress(ev, 'all_billing', 'js_change_billing');
+    async _changeBillingAddress (ev) {
+        await this._changeAddress(ev, 'all_billing', 'js_change_billing');
     },
 
     /**
      * @private
      * @param {Event} ev
      */
-    _onClickChangeShipping: function (ev) {
-        this._onClickChangeAddress(ev, 'all_shipping', 'js_change_shipping');
+    async _changeShippingAddress (ev) {
+        await this._changeAddress(ev, 'all_shipping', 'js_change_shipping');
     },
 
     /**
      * @private
      * @param {Event} ev
      */
-    _onClickChangeAddress: function (ev, rowAddrClass, cardClass) {
-        var $old = $(`.${rowAddrClass}`).find('.card.border.border-primary');
-        $old.find('.btn-addr').toggle();
-        $old.addClass(cardClass);
-        $old.removeClass('bg-primary border border-primary');
-
-        var $new = $(ev.currentTarget).parent('div.one_kanban').find('.card');
-        $new.find('.btn-addr').toggle();
-        $new.removeClass(cardClass);
-        $new.addClass('bg-primary border border-primary');
+    async _changeAddress(ev, rowAddrClass, cardClass) {
+        const oldCard = document.querySelector(`.${rowAddrClass}`).querySelector(
+            '.card.border.border-primary'
+        );
+        oldCard.classList.add(cardClass);
+        oldCard.classList.remove('bg-primary', 'border', 'border-primary');
 
-        rpc(
+        const newCard = ev.currentTarget.closest('div.one_kanban').querySelector('.card');
+        newCard.classList.remove(cardClass);
+        newCard.classList.add('bg-primary', 'border', 'border-primary');
+        const mode = newCard.getAttribute('mode');
```

Comment:
Why don't you use a dataset attribute for this? I fear that `mode` is a bit too generic and could become an HTML attribute in the future. Also, HTML with custom tags will be considered invalid by HTML validators which could result in some browsers showing the page in backward compatibility mode.  See https://stackoverflow.com/a/53464682
The same comment applies to `partner_id`.
```suggestion
        const mode = newCard.dataset.mode;
```